Userland/Libraries/LibWeb/HTML/Policy.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, Noah Bright <noah.bright.1@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashTable.h>
+#include <AK/String.h>
+#include <LibWeb/HTML/Origin.h>
+
+// https://w3c.github.io/webappsec-csp/#framework-policy
+namespace Web::HTML {
+
+// https://w3c.github.io/webappsec-csp/#directives
+struct Directive {
+    String key;
+    HashTable<String> value;
+};
+// FIXME: implement directive algorithms
+
+class Policy {
+public:
+    auto& directive_set() { return m_directive_set; }
+
+private:
+    // an ordered set
+    Vector<Directive> m_directive_set;
+
+    // either "enforce" or "report"
+    String m_disposition;
+
+    // "header" or "meta"
+    String m_source;
+
+    HTML::Origin m_self_origin;
+};
+
+// Also have serialized CSP lists - unclear to subclass or not yet
+// API here might be kind of funny with the actual vector as a member
+class CSP_List {
+public:
+    Vector<Policy>& csp_list() { return m_csp_list; }
+    bool contains_a_header_delivered_content_security_policy() const;
+    // FIXME: Implement
+    // parse_a_serialized_csp()
+    // parse_a_serialized_csp_list()
+    // Parse_response’s_Content_Security_Policies()
+
+private:
+    Vector<Policy> m_csp_list = {};
+};
+}
```

Comment:
All we've got so far is `using CSPList = Vector<Policy>`, and `bool contains_a_header_delivered_content_security_policy(CSPList const&);` as a free function.

I would rather we keep it that way until proven necessary.